crates/biome_diagnostics/src/display_github.rs

```diff
@@ -0,0 +1,107 @@
+use std::io;
+
+use biome_console::fmt;
+
+use crate::display::frame::SourceFile;
+use crate::{diagnostic::internal::AsDiagnostic, Diagnostic, Resource, Severity};
+
+/// Helper struct for printing a diagnostic as markup into any formatter
+/// implementing [biome_console::fmt::Write].
+pub struct PrintGitHubDiagnostic<'fmt, D: ?Sized> {
+    diag: &'fmt D,
+}
+
+impl<'fmt, D: AsDiagnostic + ?Sized> PrintGitHubDiagnostic<'fmt, D> {
+    pub fn simple(diag: &'fmt D) -> Self {
+        Self { diag }
+    }
+}
+
+impl<'fmt, D: AsDiagnostic + ?Sized> fmt::Display for PrintGitHubDiagnostic<'fmt, D> {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> io::Result<()> {
+        let diagnostic = self.diag.as_diagnostic();
+        let location = diagnostic.location();
+
+        // Docs:
+        // https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions
+
+        let Some(span) = location.span else {
+            return Ok(());
+        };
+        let Some(source_code) = location.source_code else {
+            return Ok(());
+        };
+
+        let file_name_unescaped = match &location.resource {
+            Some(Resource::File(file)) => file,
+            _ => return Ok(()),
+        };
+
+        let source = SourceFile::new(source_code);
+        let start = source.location(span.start())?;
+        let end = source.location(span.end())?;
+
+        let title = "TODO"; // TODO
+
+        let command = match diagnostic.severity() {
+            Severity::Error | Severity::Fatal => "error",
+            Severity::Warning => "warning",
+            Severity::Hint => "notice",
+            Severity::Information => return Ok(()),
+        };
+
```

Comment:
To print the advices, you can create a formatter right here, like in this example:

https://github.com/biomejs/biome/blob/b0cdf485f885b3a6ac4eda17bc956c113005b358/crates/biome_diagnostics/src/display.rs#L271-L272

Then, create a new struct similar to this one. This new struct will be responsible to write the message. Usually, the message:

https://github.com/biomejs/biome/blob/b0cdf485f885b3a6ac4eda17bc956c113005b358/crates/biome_diagnostics/src/display.rs#L331

Here's where we write the first message of a diagnostic. In this case we don't need to do anything special, but we should follow the same logic:

https://github.com/biomejs/biome/blob/b0cdf485f885b3a6ac4eda17bc956c113005b358/crates/biome_diagnostics/src/display.rs#L264-L293

Reply:
Do you mean using somethign like this as a buffer?
```rs
let title = {
    let mut message = String::new();
    let mut fmt = fmt::Formatter::new(&mut message);
    fmt.write_markup(markup!({ PrintDiagnosticMessage(diagnostic) }))?;
    message
};

// ...
struct PrintDiagnosticMessage<'fmt, D: ?Sized>(&'fmt D);

impl<'fmt, D: Diagnostic + ?Sized> fmt::Display for PrintDiagnosticMessage<'fmt, D> {
    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> io::Result<()> {
        let Self(diagnostic) = *self;
        diagnostic.message(fmt)?;
        Ok(())
    }
}
```

There doesn't seem to be a `biome_console::fmt::Write` implementation for `String`, so this doesn't seem to work. Or do I have to get the String from MarkupBuf somehow? Or did I get something completely wrong?

Reply:
Actually, we have a utility that we usually use in our tests/scripts:

https://github.com/biomejs/biome/blob/e3fc59d8ceed28c8df4e5d19cf80c12cda120142/xtask/lintdoc/src/main.rs#L888-L896

We can copy it and adapt it. For example, we should gracefully handle the `expect`

Reply:
I tried doing that in 2d9df8a11f7e58f3a4b02fe07e151636cf09cc7a, can you take a look?

Reply:
It looks good to me as the first implementation. I am sure there's a better way, but I need to my head into it to find a safer way :)